src/core/load-fragment.c

```diff
@@ -1416,6 +1416,98 @@ int config_parse_exec_cpu_sched_prio(const char *unit,
         return 0;
 }
 
+int config_parse_root_image_options(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(mount_options_free_allp) MountOptions *options = NULL;
+        ExecContext *c = data;
+        const char *p = rvalue;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                c->root_image_options = mount_options_free_all(c->root_image_options);
+                return 0;
+        }
+
+        for (;;) {
+                _cleanup_free_ char *partition = NULL, *mount_options = NULL, *tuple = NULL;
+                MountOptions *o = NULL;
+                unsigned int partition_number = 0;
+
+                r = extract_first_word(&p, &tuple, WHITESPACE, EXTRACT_UNQUOTE);
+                if (r == 0)
+                        break;
+                if (r == -ENOMEM)
+                        return log_oom();
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse %s: %s", lvalue, rvalue);
+                        return 0;
+                }
+
+                r = extract_first_word((const char **)&tuple, &partition, ":", EXTRACT_UNQUOTE|EXTRACT_DONT_COALESCE_SEPARATORS);
```

Comment:
this will corrupt memory. `tuple` has `_cleanup_free_` set further up, and this call will move the pointer forward, so that we'll free  apointer in the middle of a memory block, but not the pointer to the beginning of it

Reply:
I don't think we should set EXTRACT_UNQUOTE here, there's no point in quoting partition numbers

Reply:
Good catch, strange that it didn't explode when testing. Fixed.

Reg. EXTRACT_UNQUOTE, the first word could be a mount options string though, for the case where the partition number is implied, so I think it should stay? I'll rename the variables to make it a bit clearer